spock-core/src/main/java/org/spockframework/runtime/extension/builtin/RollupExtension.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.runtime.extension.builtin;
+
+import org.spockframework.runtime.InvalidSpecException;
+import org.spockframework.runtime.extension.AbstractAnnotationDrivenExtension;
+import org.spockframework.runtime.extension.AbstractGlobalExtension;
+import org.spockframework.runtime.model.FeatureInfo;
+import org.spockframework.runtime.model.SpecInfo;
+import spock.lang.Rollup;
+import spock.lang.Unroll;
+
+public class RollupExtension extends AbstractAnnotationDrivenExtension<Rollup> {
+  @Override
+  public void visitSpecAnnotation(Rollup rollup, SpecInfo spec) {
+    visitSpecAnnotation(spec, true);
+  }
+
+  private void visitSpecAnnotation(SpecInfo spec, boolean direct) {
+    if (direct && (spec.getAnnotation(Unroll.class) != null)) {
+      throw new InvalidSpecException("@Unroll and @Rollup must not be used on the same element");
```

Comment:
That exception is repeated 4 times. Maybe there could be a separate exception class created to simplify it to `throw new InvalidUnrollingConfigurationException()` (name to be possibly improved to cover both `*Rolling`)?

Reply:
Well, I did it like for other cases, for example if a `@ClassRule` is not also `@Shared` which also throws an `InvalidSpecException`.
I could make a `InvalidUnrollingConfigurationException` as subclass of `InvalidSpecException`, or I could make some method that produces the exception, for example `UnrollConfiguration#invalidAnnotationCombination()` that either directly throws the `InvalidSpecException`, or returns the exception to be thrown.
@leonard84 what do you think about this?

Reply:
With the rewritten extensions it is only two occurrences left and with different messages, so I consider this as resolved.